src/dash-playlist-loader.js

```diff
@@ -134,48 +228,166 @@ export default class DashPlaylistLoader extends EventTarget {
-            });
+      this.masterXml_ = req.responseText;
+
+      if (req.responseHeaders && req.responseHeaders.date) {
+        this.masterLoaded_ = Date.parse(req.responseHeaders.date);
+      } else {
+        this.masterLoaded_ = Date.now();
+      }
+
+      this.syncClientServerClock_(this.onClientServerClockSync_.bind(this));
+    });
+  }
+
+  syncClientServerClock_(done) {
+    const utcTiming = mpdParser.parseUTCTiming(this.masterXml_);
+
+    // No UTCTiming element found in the mpd. Use Date header from mpd request as the
+    // server clock
+    if (utcTiming === null) {
+      this.clientOffset_ = this.masterLoaded_ - Date.now();
+      return done();
+    }
+
+    if (utcTiming.method === 'DIRECT') {
+      this.clientOffset_ = utcTiming.value - Date.now();
+      return done();
+    }
+
+    this.request = this.hls_.xhr({
+      uri: resolveUrl(this.srcUrl, utcTiming.value),
+      method: utcTiming.method,
+      withCredentials: this.withCredentials
+    }, (error, req) => {
+      // disposed
+      if (!this.request) {
+        return;
+      }
+
+      if (error) {
+        // sync request failed, fall back to using date header from mpd
+        // TODO: log warning
+        this.clientOffset_ = this.masterLoaded_ - Date.now();
+        return done();
+      }
+
+      let serverTime;
+
+      if (utcTiming.method === 'HEAD') {
+        if (!req.responseHeaders || !req.responseHeaders.date) {
+          // expected date header not preset, fall back to using date header from mpd
+          // TODO: log warning
+          serverTime = this.masterLoaded_;
+        } else {
+          serverTime = Date.parse(req.responseHeaders.date);
         }
+      } else {
+        serverTime = Date.parse(req.responseText);
       }
 
-      // set up phony URIs for the playlists since we won't have external URIs for DASH
-      // but reference playlists by their URI throughout the project
-      for (let i = 0; i < this.master.playlists.length; i++) {
-        const phonyUri = `placeholder-uri-${i}`;
+      this.clientOffset_ = serverTime - Date.now();
+
+      done();
+    });
+  }
+
+  onClientServerClockSync_() {
+    this.master = this.parseMasterXml();
+
+    this.state = 'HAVE_MASTER';
 
-        this.master.playlists[i].uri = phonyUri;
-        // set up by URI references
-        this.master.playlists[phonyUri] = this.master.playlists[i];
+    this.trigger('loadedplaylist');
+
+    if (!this.media_) {
+      // no media playlist was specifically selected so start
+      // from the first listed one
+      this.media(this.master.playlists[0]);
+    }
+    // trigger loadedmetadata to resolve setup of media groups
+    // trigger async to mimic behavior of HLS, where it must request a playlist
+    setTimeout(() => {
+      this.trigger('loadedmetadata');
+    }, 0);
+
+    if (this.master.minimumUpdatePeriod) {
+      setTimeout(() => {
+        this.trigger('minimumUpdatePeriod');
+      }, this.master.minimumUpdatePeriod);
```

Comment:
There is one special case here, where the minimumUpdatePeriod = 0.  Which effectively means "update this as often as you want".  Is this case covered?

Reply:
That case is not covered by this, I can add a TODO making note of that? I feel like for the scope of this PR we can ignore that case for now until it comes up. I think the best thing to do in this scenario would be to refresh the xml at the same rate we "refresh" the media playlist, i.e. targetDuration

Reply:
TODO seems good.